### ISSUE TITLE
FormGroup has many links incorrectly going to AbstractControl methods

### ISSUE THREAD

**Issue by perrierism** [angular/angular#19387]
## I'm submitting a...
<pre><code>
[ X] Documentation issue or request
</code></pre>

## Current behavior
In [FormGroup: Overview](https://angular.io/api/forms/FormGroup#overview), the links for most methods incorrectly point to AbstractControl methods of the same name. Appears to be a copypasta error on all links at or below "`validator: ValidatorFn|null`". Many of the method signatures for these links are also not right for the FormGroup methods (`setValue` for instance) .

## Expected behavior
The links should go to the methods further down the page which have specific signatures for FormGroup.

Any necessary links to AbstractControl (any?) should be presented in a way that makes it clear that user is being taken to a different class / page.

## Minimal reproduction of the problem with instructions
1. [FormGroup: Overview](https://angular.io/api/forms/FormGroup#overview)
2. Click many of the methods in the "**Overview**" section, e.g., "setValue" 

**Result:** Link goes to **AbstractControl** documentation for method of same name instead of the FormGroup specific documentation further down the page.

**Also note:** The signatures shown are incorrect for FormGroup (AbstractControl signatures are shown)

## What is the motivation / use case for changing the behavior?
Anti-chaos sentiment.

## Environment
https://angular.io/api/forms/FormGroup on 9/25/17

**Comment by juristr** [comment]
@perrierism Hi, I'm not quite sure this is actually wrong. The overview you're talking about gets generated automatically out of the underlying source class. There are duplicated functions in there, one for the actuall class `FormName` and another for the `AbstractControl` it inherits from. Take for instance `setValue(...)`.

If at all, maybe the 2nd `setValue(...)` from the `AbstractControl` shouldn't be in there. What do you think @wardbell? Should the [class-overview](https://github.com/angular/angular/blob/master/aio/tools/transforms/templates/api/includes/class-overview.html) template and according [memberHelper](https://github.com/angular/angular/blob/master/aio/tools/transforms/templates/api/lib/memberHelpers.html) be tuned for this?

**Comment by juristr** [comment]
Oh, by looking through the code that renders this class I found that it also writes out a comment about this in the overview 😅...which I actually didn't see before:

![image](https://user-images.githubusercontent.com/542458/30846528-52b9c938-a298-11e7-8a75-613990931c59.png)

tbh, a bit confusing, yes 🤔

**Comment by perrierism** [comment]
Hmm yes. Possible easy fix could be to just split the overview code/example section into with a short explanation in plain text rather than with a comment in the code. Something like


# Overview

```
class FormGroup extends AbstractControl {
  constructor(controls: {[key: string]: AbstractControl}, validator?: ValidatorFn|null, asyncValidator?: AsyncValidatorFn|null)
  controls: {[key: string]: AbstractControl}
  registerControl(name: string, control: AbstractControl): AbstractControl
...
```
FormGroup Methods inherited from AbstractControl
```
 validator: ValidatorFn|null
  asyncValidator: AsyncValidatorFn|null
  get value(): any
  get parent(): FormGroup|FormArray
  get status(): string
...
```

This would make it more clear IMO. I'm wondering though if `setValue()` should be listed as an inherited method? It is one that is defined on FormGroup and actually takes different parameters.

**Comment by jenniferfell** [comment]
@petebacondarwin Is this still a problem, or resolved by new templates?

**Comment by petebacondarwin** [comment]
It is not yet resolved...

**Comment by petebacondarwin** [comment]
It should be resolved by #25768 I believe

**Comment by petebacondarwin** [comment]
https://github.com/angular/angular/pull/25768 has now landed.
